CONTRIBUTING.md

```diff
@@ -0,0 +1,128 @@
+with a high-level proposal and discuss it with the core contributors through
+Github comments or in [Slack][slack]. After reaching a consensus with core
+contributors about the change, discuss the best way to go about implementing it.
+
+> Tip: Keep your master branch pointing at the original repository and make
+> pull requests from branches on your fork. To do this, run:
+>
+> ```sh
+> git remote add upstream https://github.com/godaddy/<this-project>.git
+> git fetch upstream
+> git branch --set-upstream-to=upstream/master master
+> ```
+>
+> This will add the original repository as a "remote" called "upstream," Then
+> fetch the git information from that remote, then set your local master
+> branch to use the upstream master branch whenever you run git pull. Then you
+> can make all of your pull request branches based on this master branch.
+> Whenever you want to update your version of master, do a regular git pull.
+
+## Code Review
+
+Any open source project relies heavily on code review to improve software
+quality. All significant changes, by all developers, must be reviewed before
+they are committed to the repository. Code reviews are conducted on GitHub
+through comments on pull requests or commits. The developer responsible for a
+code change is also responsible for making all necessary review-related changes.
+
+Sometimes code reviews will take longer than you would hope for, especially for
+larger features. Here are some accepted ways to speed up review times for your
+patches:
+
+- Review other people’s changes. If you help out, others will more likely be
+  willing to do the same for you.
+- Split your change into multiple smaller changes. The smaller your change,
+  the higher the probability that somebody will take a quick look at it.
+- Mention the change on [Slack][slack]. If it is urgent, provide reasons why it
+  is important to get this change landed. Remember that you are asking for
+  valuable time from other professional developers.
+
+**Note that anyone is welcome to review and give feedback on a change, but only
+people with commit access to the repository can approve it.**
+
+## Attribution of Changes
+
+When contributors submit a change to this project, after that change is
+approved, other developers with commit access may commit it for the author. When
+doing so, it is important to retain correct attribution of the contribution.
+Generally speaking, Git handles attribution automatically.
+
+## Code Style and Documentation
+
+Ensure that your contribution follows the standards set by the project's style
+guide with respect to patterns, naming, documentation and testing.
+
+## Additional Resources
+
+- [General GitHub Documentation](https://help.github.com/)
+- [GitHub Pull Request documentation](https://help.github.com/send-pull-requests/)
+
+[coc]: https://github.com/godaddy/.github/blob/master/CODE_OF_CONDUCT.md
+[security]: https://github.com/godaddy/.github/blob/master/SECURITY.md
```

Comment:
Trying to remember how these show in a repo.  What happens if a repo has their own CoC or SECURITY file but not a CONTRIBUTING file.  Is there a way to link to those from this generic document?

Reply:
If somebody goes into the "Community" tab, and clicks on "Contributing", it will go directly to this doc, in this repo. So, I don't think this is something which we can really account for. I also think that is a somewhat of a corner case, and really more of an opportunity for us to encourage that repo to adopt their own contributing doc, as opposed to working hard to account for.